src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -76,16 +78,22 @@ public async Task Subscribe(Subscriber subscriber, MessageType messageType, Cont
         public async Task Unsubscribe(Subscriber subscriber, MessageType messageType, ContextBag context)
         {
             var table = client.GetTableReference(subscriptionTableName);
-            var encodedAddress = EncodeTo64(subscriber.TransportAddress);
-
-            var retrieveOperation = TableOperation.Retrieve<TimeoutDataEntity>(messageType.ToString(), encodedAddress);
```

Comment:
This is very weird. I think it conincedently worked because we do not really care about the shape of the data but just about the partition key and row key. Still I don't think it helps to load and then delete when we can directly delete and just ignore any NotFound issue